core/src/main/scala/org/broadinstitute/dsde/rawls/dataaccess/slick/WorkspaceComponent.scala

```diff
@@ -376,28 +376,19 @@ trait WorkspaceComponent {
       filter(rec => (rec.namespace === namespaceName.value))
     }
 
-    def listOutdatedEntityCaches(limit: Int): ReadAction[Seq[(UUID, Timestamp)]] = {
-      //We don't necessarily want all of the workspaces back, just the ones that we can handle in one pass
-      //We also order by lastModified so we can handle the ones that are the longest out of date
-      //0 is a "magic" timestamp that means that the cache should not be updated because it produces
-      //an error that breaks the statistics cache monitor. These are unexpected cases that would
-      //need manual intervention.
-      filter(rec => (rec.entityCacheLastUpdated < rec.lastModified && rec.entityCacheLastUpdated > new Timestamp(0))).take(limit).sortBy(_.lastModified.asc).map { ws => (ws.id, ws.lastModified) }.result
+    def findMostOutdatedEntityCacheAfter(timestamp: Timestamp): ReadAction[Option[(UUID, Timestamp)]] = {
+      // Find the workspace that has the entity cache that is the most out of date:
+      // A. Workspace has a cacheLastUpdated date that is not current ("current" means equal to lastModified)
+      // B. cacheLastUpdated is after @param timestamp
+      // C. Ordered by lastModified from oldest to newest. Meaning, return the workspace that was modified the longest ago
+      uniqueResult[(UUID, Timestamp)](filter(rec => rec.entityCacheLastUpdated < rec.lastModified && rec.entityCacheLastUpdated > timestamp).sortBy(_.lastModified.asc).take(1).map { ws => (ws.id, ws.lastModified) })
     }
 
-    def listBackloggedEntityCaches(limit: Int): ReadAction[Seq[(UUID, Timestamp)]] = {
-      filter(rec => rec.entityCacheLastUpdated.isEmpty).sortBy(_.lastModified.desc).take(limit).map { ws => (ws.id, ws.lastModified) }.result
-    }
-
-    //Three modes:
-    // Some(true): cache exists and is up to date
-    // Some(false) :cache exists and is not up to date
-    // None: cache does not exist
-    def isEntityCacheCurrent(workspaceId: UUID): ReadAction[Option[Boolean]] = {
-      val queryResult = uniqueResult[(Timestamp, Option[Timestamp])](filter(rec => rec.id === workspaceId).map(ws => (ws.lastModified, ws.entityCacheLastUpdated)))
+    def isEntityCacheCurrent(workspaceId: UUID): ReadAction[Boolean] = {
+      val queryResult = uniqueResult[(Timestamp, Timestamp)](filter(rec => rec.id === workspaceId).map(ws => (ws.lastModified, ws.entityCacheLastUpdated)))
       queryResult.map {
-        case Some((lastModified, Some(entityCacheLastUpdated))) => Option(lastModified.equals(entityCacheLastUpdated))
-        case _ => None
+        case Some((lastModified, entityCacheLastUpdated)) => lastModified.equals(entityCacheLastUpdated)
+        case _ => false
```

Comment:
I think there would still need to be a match somewhere even if I pushed that logic into the query because I'd still need to pass this through `uniqueResult` to process it as one result